### PR TITLE
CDI-646 Annotated configurators

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

**Comment by manovotn** [comment]
After convincing Martin to keep `removeAll` methods, this has my +1 :)

File: api/src/main/java/javax/enterprise/inject/spi/configurator/AnnotatedTypeConfigurator.java [hunk 1]
```diff
@@ -53,33 +61,32 @@
...
      *
-     * @param annotationType annotation class to remove
+     * @param annotationType the annotation type
      * @return self
      */
     AnnotatedTypeConfigurator<T> remove(Class<? extends Annotation> annotationType);
 
     /**
-     * Remove all annotations from the type.
+     * Remove all the annotations from the type.
```
Comment (Emily-Jiang) [inline api/src/main/java/javax/enterprise/inject/spi/configurator/AnnotatedTypeConfigurator.java]: all the - not grammatically correct. Revert change is better.

**Comment by tremes** [comment]
Yes using predicates is nice idea. +1

File: spec/src/main/asciidoc/core/spi.asciidoc [hunk 2]
```diff
@@ -847,13 +845,12 @@ public interface AnnotatedTypeConfigurator<T> {
 
 * `getAnnotated()` returns the original `AnnotatedType` with which this configurator was initialized
 * `add()` adds an annotation to the configured `AnnotatedType`
-* `remove()` methods remove all annotation from the configured `AnnotatedType`
-* `removeAll()` remove all existing annotation on the configured `AnnotatedType`
```
Comment (manovotn) [inline spec/src/main/asciidoc/core/spi.asciidoc]: Just noticed this - since we have kept the `removeAll` method, this line should stay, right?

Comment (mkouba) [inline spec/src/main/asciidoc/core/spi.asciidoc]: Yep, thanks for noticing.

**Pull request merged**